app/controllers/application_controller/buttons.rb

```diff
@@ -250,6 +251,17 @@ def applies_to_class_model(applies_to_class)
     applies_to_class == "ServiceTemplate" ? Service : applies_to_class.constantize
   end
 
+  def custom_button_done
+    url = SystemConsole.where(:vm => params[:id]).first.try(:url)
```

Comment:
@martinpovolny 
Will we have a 'ur'l as a custom attribute available for vm objects which the Automate Method would set?